python_basics_writing_python_code/python_basics_writing_python_code.md

```diff
@@ -903,35 +856,72 @@ print(contact_info_copy)
     </div>
 
-* There isn't a method to add items to a dictionary, but it can be done! You can give the dictionary a new key and assign a value to it (notice that we're using bracket notation again). This is another example of a permanent change to our dictionary.
+* There isn't a method to add items to a dictionary, but it can be done! You can give the dictionary a new key and assign a value to it (notice that we're using square bracket notation again, like we did for lists). This is another example of a permanent change to our dictionary.
 
     @sage
     <div class="python_link">
     <lia-keep>
     <script type="text/x-sage">
 
     contact_info["work_email"] = "myname@company.com"
-print(contact_info)
+    print(contact_info)
 
     </script>
     </lia-keep>
     </div>
 
   You can also use the same syntax with an existing key to assign a new value (note that the new value will **replace** the previous value).
 
-### Quiz: Dictionaries
+<div class = "help">
+<b style="color: rgb(var(--color-highlight));">Troubleshooting help</b><br>
 
-True or False: Duplicate key-value pairs are allowed in dictionaries.
+You might have noticed by now that we have used square brackets in a few different contexts, and it can be difficult to keep straight! Here are the two uses you need to know for this module:
 
-[( )] True
-[(X)] False
-***
-<div class = "answer">
+1. Creating lists: You use square brackets to create a list object (either an empty list or one that has items in it).
 
-While **values** can be repeated in a dictionary, each **key** must be unique. However, you can have multiple values for a key by putting those values in a list.
+2. Accessing items in a collection: This is the subsetting notation that we talked about earlier, and it used to retrieve an item in a list or a dictionary. You do this by using the index in a list and by the key in a dictionary. 
 
 </div>
-***
+
+### Quiz: Dictionaries
+
+1. True or False: Duplicate key-value pairs are allowed in dictionaries.
+
+    [( )] True
+    [(X)] False
+    ***
+    <div class = "answer">
+
+    While **values** can be repeated in a dictionary, each **key** must be unique. However, you can have multiple values for a key by putting those values in a list.
+
+    </div>
+    ***
+
+2. How would you access all of the counties (the "keys") in the dictionary below? 
+
+    `capital_cities = {"Afghanistan" : "Kabul", "Albania" : "Tirana", "Algeria" : "Algiers", "Andorra" : "Andorra la Vella"}`
+
+    [[capital_cities.keys() ]]
+    ***
+    <div class = "answer">
+
+    To return the keys of a dictionary, you use the `.keys()` dictionary method.  
+
+    </div>
+    ***
+
+3. How would you add the the country of Angola and its capital city of Luanda to the dictionary below?
+
+    `capital_cities = {"Afghanistan" : "Kabul", "Albania" : "Tirana", "Algeria" : "Algiers", "Andorra" : "Andorra la Vella"}`
+
+    [[capital_cities["Angola"] = "Luanda"]]
```

Comment:
```suggestion
    [[capital_cities["Angola"] = "Luanda"]]
    <script>
  let input = "@input".replace(/\s+/g, '');
  input == "capital_cities[\"Angola\"]=\"Luanda\"";
</script>
```

I think this will work to allow for a difference in the number of spaced (e.g. I tried with no spaces around the `=`)

Reply:
Oooh, good call!